lib/install/validate/network.go

```diff
@@ -74,40 +82,157 @@ func (v *Validator) checkNetworkConflict(bridgeNetName, otherNetName, otherNetTy
+// prevent assigning > 1 static IP to the same port group
+func (v *Validator) checkPortGroups(counts map[string]int, ips map[string][]data.NetworkConfig) error {
+	defer trace.End(trace.Begin(""))
+
+	for pg := range ips {
+		if len(ips[pg]) > 1 {
+			var msgIPs []string
+			for _, v := range ips[pg] {
+				msgIPs = append(msgIPs, v.IP.IP.String())
+			}
+			log.Errorf("Port group %q is configured for networks with more than one static IP: %s", pg, msgIPs)
+			log.Error("All VCH networks on the same port group must have the same IP address")
+			log.Error("To resolve this, configure static IP for one network and assign other networks to same port group.")
+			log.Error("The static IP will be automatically configured for networks sharing the port group.")
+			return fmt.Errorf("Incorrect static IP configuration for networks on port group %q", pg)
+		}
+	}
+	return nil
+}
+
+// configureSharedPortGroups sets VCH static IP for networks that share a
+// portgroup with another network that has a configured static IP
+func (v *Validator) configureSharedPortGroups(input *data.Data, counts map[string]int, ips map[string][]data.NetworkConfig) error {
+	defer trace.End(trace.Begin(""))
+
+	// find other networks using same portgroup and copy the NetworkConfig to them
+	for name, config := range ips {
+		if len(config) != 1 {
+			return fmt.Errorf("Failed to configure static IP for additional networks using port group %q", name)
+		}
+		log.Infof("Configuring static IP for additional networks using port group %q", name)
+		if input.ClientNetwork.Name == name && input.ClientNetwork.Empty() {
+			input.ClientNetwork = config[0]
+		}
+		if input.ExternalNetwork.Name == name && input.ExternalNetwork.Empty() {
+			input.ExternalNetwork = config[0]
+		}
+		if input.ManagementNetwork.Name == name && input.ManagementNetwork.Empty() {
+			input.ManagementNetwork = config[0]
+		}
+	}
+
+	return nil
+}
+
 func (v *Validator) network(ctx context.Context, input *data.Data, conf *config.VirtualContainerHostConfigSpec) {
 	defer trace.End(trace.Begin(""))
 
+	var e *executor.NetworkEndpoint
+	var err error
+
+	// set default portgroup if user input not provided
+	if input.ClientNetwork.Name == "" {
+		input.ClientNetwork.Name = input.ExternalNetwork.Name
+	}
+	if input.ManagementNetwork.Name == "" {
+		input.ManagementNetwork.Name = input.ClientNetwork.Name
+	}
+
+	c := make(map[string]int)                  // number of VCH networks using portgroup
+	i := make(map[string][]data.NetworkConfig) // user configured IPs for portgroup
+	v.portGroupConfig(input, c, i)
+
+	err = v.checkPortGroups(c, i)
+	if err != nil {
+		v.NoteIssue(err)
+	}
+
+	err = v.configureSharedPortGroups(input, c, i)
+	if err != nil {
+		v.NoteIssue(err)
+	}
+
 	// External net
 	// external network is default for appliance
-	err := v.addNetworkHelper(ctx, conf, input.ExternalNetworkName, "external", "external", true)
+	if !input.ExternalNetwork.Empty() {
```

Comment:
Can we collapse this to a common call of:
```
	e, err = v.getEndpoint(ctx, conf, input.ExternalNetwork.Name, "external", "external", true, input.VCHDNS)
```

and move the log message and check for Empty/Static into getEndpoint? We're unpacking every element of `NetworkConfig` in the arguments to `getStaticEndpoint` and I don't see it being called from anywhere else.

IMO this would make this portion of the code significantly more readable